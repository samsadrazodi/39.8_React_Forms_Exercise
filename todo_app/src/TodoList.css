/* TodoList.css */

.TodoList {
    font-family: 'Roboto', sans-serif;
    margin: 0 auto;
    padding: 20px;
    width: 400px;
    background-color: #dfdcdca8;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  }
  
  .TodoList h3 {
    font-size: 1.5rem;
    margin-bottom: 15px;
    color: #121111;
  }
  
  .NewTodoForm {
    margin-bottom: 20px;
  }
  
  ul {
    list-style: none;
    padding: 0;
  }
  
  li {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    margin-bottom: 10px;
    background-color: #fff;
    border-radius: 5px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: background-color 0.2s ease, transform 0.2s ease;
  }
  
  li:hover {
    background-color: #f9f9f9;
    transform: translateX(5px);
  }
  
  li input[type="text"] {
    flex: 1;
    padding: 8px;
    border: none;
    border-radius: 3px;
    background-color: #f5f5f5;
    color: #333;
    font-size: 14px;
  }
  
  li button {
    background-color: #4caf50;
    color: white;
    border: none;
    border-radius: 3px;
    padding: 5px 10px;
    cursor: pointer;
    transition: background-color 0.2s ease;
  }
  
  li button:hover {
    background-color: #45a049;
  }
  
  .Todo {
    flex-grow: 1;
    padding-left: 10px;
    color: #333;
  }
  
  .TodoButtons {
    display: flex;
    align-items: center;
  }
  
  .TodoButtons button {
    background-color: #f44336;
    color: white;
    border: none;
    border-radius: 3px;
    margin-left: 5px;
    padding: 3px 7px;
    cursor: pointer;
    transition: background-color 0.2s ease;
  }
  
  .TodoButtons button.editButton {
    background-color: #2196f3;
  }
  
  .TodoButtons button:hover {
    background-color: #e53935;
  }
  