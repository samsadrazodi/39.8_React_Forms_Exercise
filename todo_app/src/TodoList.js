import React, { useState } from "react";
import './TodoList.css';
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const INITIAL_STATE = [

    ];
    const [todos, setTodos] = useState(INITIAL_STATE);
    const [editIndex, setEditIndex] = useState(-1);
    const [editedTodo, setEditedTodo] = useState(""); // State to hold edited todo text

    const handleRemove = (value) => {
        setTodos(todos.filter((todo) => todo.todo !== value));
    }

    const handleEdit = (index, todo) => {
        setEditIndex(index);
        setEditedTodo(todo); // Initialize the edited todo text
    }

    const handleUpdate = (index) => {
        const updatedTodos = [...todos];
        updatedTodos[index].todo = editedTodo;
        setTodos(updatedTodos);
        setEditIndex(-1);
    }

    const addTask = (newTask) => {
        setTodos(items => [...items, { ...newTask }]);
    }

    return (
        <div className="TodoList">
            <h3>TO DO LIST:</h3>
            <NewTodoForm addTask={addTask} />


            <ul>
                {todos.map(({ todo }, index) => (
                    <li key={index}>
                        {editIndex === index ? (
                            <>
                                <input
                                    type="text"
                                    value={editedTodo}
                                    onChange={(e) => setEditedTodo(e.target.value)}
                                />
                                <button onClick={() => handleUpdate(index)}>Save</button>
                            </>
                        ) :( <Todo  todo={todo} handleRemove={handleRemove} handleEdit={handleEdit} index={index} /> )
                           
                        }
                    </li>
                ))}
            </ul>
            
        </div>
    );
}

export default TodoList;
