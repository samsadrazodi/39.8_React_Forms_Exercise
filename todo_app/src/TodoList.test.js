import React from "react";
import {render, fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";


it('renders without crashing', function (){
    render(<TodoList />);
});

it('matches snapshot', function (){
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('should add a new Todo', function (){
    const {queryByText, getByLabelText} = render(<TodoList />);
    const input = getByLabelText('Todo Name:');
    const btn = queryByText('Add Todo');

    fireEvent.change(input,{target:{value:'test the app'}});
    fireEvent.click(btn);

    expect(queryByText('test the app')).toBeInTheDocument();
})