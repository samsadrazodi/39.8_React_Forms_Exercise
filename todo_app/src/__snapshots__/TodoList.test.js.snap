// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h3>
      TO DO LIST:
    </h3>
    <form>
      <label
        for="todo"
      >
        Todo Name:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Todo Name"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
